### PR TITLE
DAS-103: digitize USACE CBR curves v1 + PCHIP interpolation utility

### PR DESCRIPTION
**Pull request by rachmad-jenss**
## Summary

- Digitize USACE TM 5-822-12 Figure 1 CBR design curves as versioned JSON (`usace_cbr_v1.json`) with 9 coverage levels (10–100,000) and 14 CBR knot points (2–100%)
- Add provenance metadata companion file (`usace_cbr_v1.meta.yaml`) per Curve Encoding Policy
- Implement `src/haulpave/utils/interpolation.py` with `load_curve_data()`, `_validate_curve_data()`, and `interpolate_thickness()` using 2-step PCHIP interpolation (CBR axis then log-coverage axis)
- Add 16 unit tests covering benchmark values, schema validation, monotonicity, boundary behaviour, coverage clamping, and error cases
- Update `design/curves/README.md` to mark curve files as Implemented

## Test plan

- [x] `load_curve_data("usace_cbr_v1")` returns dict with all expected schema keys
- [x] `interpolate_thickness(data, cbr=10, coverages=1000)` returns ~408 mm (±5 mm)
- [x] `interpolate_thickness(data, cbr=5, coverages=100)` returns ~462 mm (±5 mm)
- [x] Monotonicity: higher CBR → lower thickness at constant coverages
- [x] Monotonicity: higher coverages → higher thickness at constant CBR
- [x] `ValueError` raised for CBR=1 (below range)
- [x] `ValueError` raised for CBR=101 (above range)
- [x] CBR=2 and CBR=100 boundary values work without error
- [x] `ValueError` raised for coverages=0
- [x] `ValueError` raised for coverages=-1
- [x] `pytest tests/ --cov=haulpave --cov-fail-under=85` passes (97.67% coverage, 66 tests)
- [x] `ruff check` and `mypy` pass with no issues

### CODE REVIEW

File: src/haulpave/design/curves/usace_cbr_v1.meta.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+digitization_method: "Manual point extraction from published figure, interpolated at standard CBR knot points"
+digitization_date: "2026-03-24"
+digitized_by: "HaulPave project team"
+tolerance_mm: 5
+notes: >
+  Values are representative digitizations of the published USACE curves.
+  Thickness represents total required pavement thickness above subgrade (surfacing + base + subbase).
+  CBR values are in-situ subgrade CBR (%).
+  Design coverages represent equivalent design wheel load passes.
+  For CBR < 2 or CBR > 100, extrapolation is not recommended — use boundary values.
```
[the remark this answers is not in the document]
Comment (rachmad-jenss) [reply src/haulpave/design/curves/usace_cbr_v1.meta.yaml]: Fixed — updated  in  to correctly state that out-of-range CBR raises ValueError, and that out-of-range coverages are clamped to the boundary.

File: src/haulpave/utils/interpolation.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    coverage_levels = np.array(curve_data["coverage_levels"], dtype=float)
+
+    if cbr < cbr_values[0] or cbr > cbr_values[-1]:
+        raise ValueError(f"CBR {cbr} outside supported range [{cbr_values[0]}, {cbr_values[-1]}]")
+    if coverages <= 0:
+        raise ValueError(f"coverages must be > 0, got {coverages}")
+
+    # Clamp coverages to range
+    log_cov = np.log10(np.clip(coverages, coverage_levels[0], coverage_levels[-1]))
+    log_cov_levels = np.log10(coverage_levels)
```
[the remark this answers is not in the document]
Comment (rachmad-jenss) [reply src/haulpave/utils/interpolation.py:96]: Fixed — updated the docstring to clarify that positive out-of-range coverages are clamped to the boundary (not a ValueError), while coverages <= 0 still raises ValueError.

File: src/haulpave/utils/interpolation.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    # For each coverage level, interpolate thickness at the requested CBR
+    thicknesses_at_coverages: list[float] = []
+    for cov_key in curve_data["coverage_levels"]:
+        t_arr = np.array(curve_data["thickness_mm"][str(cov_key)], dtype=float)
+        pchip = PchipInterpolator(cbr_values, t_arr)
+        thicknesses_at_coverages.append(float(pchip(cbr)))
```
[the remark this answers is not in the document]
Comment (rachmad-jenss) [reply src/haulpave/utils/interpolation.py:103]: Fixed — added _validate_curve_data() helper that checks required keys, minimum axis lengths, matching coverage keys, and per-coverage array length alignment. Called at the start of interpolate_thickness().

File: tests/test_utils/test_interpolation.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    def test_raises_value_error_zero_coverages(self, curve_data: dict) -> None:  # type: ignore[type-arg]
+        """Zero coverages raises ValueError."""
+        with pytest.raises(ValueError, match="coverages"):
+            interpolate_thickness(curve_data, cbr=10, coverages=0)
+
+    def test_raises_value_error_negative_coverages(self, curve_data: dict) -> None:  # type: ignore[type-arg]
+        """Negative coverages raises ValueError."""
+        with pytest.raises(ValueError, match="coverages"):
+            interpolate_thickness(curve_data, cbr=10, coverages=-1)
```
[the remark this answers is not in the document]
Comment (rachmad-jenss) [reply tests/test_utils/test_interpolation.py:143]: Fixed — added test_coverages_below_min_clamps_to_min_level and test_coverages_above_max_clamps_to_max_level, plus three tests for _validate_curve_data (missing keys, length mismatch, missing coverage key).
